spark/v3.4/spark/src/test/java/org/apache/iceberg/spark/actions/TestRemoveOrphanFilesAction.java

```diff
@@ -450,7 +451,12 @@ public void testManyTopLevelPartitions() throws InterruptedException {
 
     Dataset<Row> df = spark.createDataFrame(records, ThreeColumnRecord.class);
 
-    df.select("c1", "c2", "c3").write().format("iceberg").mode("append").save(tableLocation);
+    df.select("c1", "c2", "c3")
+        .write()
+        .format("iceberg")
+        .option(SparkWriteOptions.USE_TABLE_DISTRIBUTION_AND_ORDERING, "false")
```

Comment:
Why do we need to turn this off for the test?

Reply:
The check below compares rows, which started to fail. Before this PR, the incoming data was not ordered as the table had a truncate transform (unsupported without extensions). After this PR, there is a sort, which breaks the test.

Reply:
We could compare counts too. No preference on my side so I went for the smallest change.

Reply:
I would probably just compare counts, otherwise it makes it seem like there is some relationship between Distribution and Ordering and this test.

Reply:
Sound good, I'll switch.

Reply:
Switched.